MIVisionX-setup.py

```diff
@@ -246,28 +303,46 @@
         os.system('sudo -v')
         os.system('(cd '+deps_dir+'/protobuf-'+ProtoBufVersion +
                   '; sudo '+linuxFlag+' ldconfig )')
+
     if neuralNetInstall == 'yes':
+        # Remove Previous Install - MIOpen
+        os.system('sudo -v')
+        if os.path.exists(ROCM_PATH+'/miopen'):
+            os.system('sudo rm -rf '+ROCM_PATH+'/miopen*')
+        if backend == 'OCL':
+            os.system('sudo '+linuxFlag+' '+linuxSystemInstall+' ' +
+                      linuxSystemInstall_check+' remove miopen-hip')
+        else:
+            os.system('sudo '+linuxFlag+' '+linuxSystemInstall+' ' +
+                      linuxSystemInstall_check+' remove miopen-opencl')
+
         os.system('(cd '+deps_dir+'/build; mkdir rocm-cmake MIOpenGEMM MIOpen)')
         # Install ROCm-CMake
         os.system('(cd '+deps_dir+'/build/rocm-cmake; ' +
                   linuxCMake+' ../../rocm-cmake )')
         os.system('(cd '+deps_dir+'/build/rocm-cmake; make -j8 )')
         os.system('(cd '+deps_dir+'/build/rocm-cmake; sudo ' +
                   linuxFlag+' make install )')
-        # Install MIOpenGEMM
-        os.system('(cd '+deps_dir+'/build/MIOpenGEMM; '+linuxCMake +
-                  ' ../../MIOpenGEMM-'+MIOpenGEMMVersion+' )')
-        os.system('(cd '+deps_dir+'/build/MIOpenGEMM; make -j8 )')
-        os.system('(cd '+deps_dir+'/build/MIOpenGEMM; sudo ' +
-                  linuxFlag+' make install )')
+        if backend == 'OCL':
+            # Install MIOpenGEMM
+            os.system('(cd '+deps_dir+'/build/MIOpenGEMM; '+linuxCMake +
+                      ' ../../MIOpenGEMM-'+MIOpenGEMMVersion+' )')
+            os.system('(cd '+deps_dir+'/build/MIOpenGEMM; make -j8 )')
+            os.system('(cd '+deps_dir+'/build/MIOpenGEMM; sudo ' +
+                      linuxFlag+' make install )')
         # Install MIOpen
         os.system('(cd '+deps_dir+'/MIOpen-'+MIOpenVersion+'; sudo ' +
                   linuxFlag+' '+linuxCMake+' -P install_deps.cmake --minimum )')
-        os.system('(cd '+deps_dir+'/build/MIOpen; '+linuxCMake +
-                  ' -DMIOPEN_BACKEND=OpenCL -DMIOPEN_USE_MIOPENGEMM=On ../../MIOpen-'+MIOpenVersion+' )')
+        if backend == 'OCL':
+            os.system('(cd '+deps_dir+'/build/MIOpen; '+linuxCMake +
```

Comment:
to build MIOpen with HIP backend, we would need to set the C++ compiler to clang++ in the cmake command.
add **CXX=/opt/rocm/llvm/bin/clang++** before **'+linuxCMake +** in line 337